src/cloudWatchLogs/timeFilterSubmenu.ts

```diff
@@ -112,7 +112,7 @@ export class TimeFilterSubmenu extends Prompter<TimeFilterResponse> {
                     break
                 }
                 case 'custom-range': {
-                    const resp = await this.customPrompter.prompt()
+                    const resp = await this.createDateBox().prompt()
```

Comment:
Looking at the rest of this code, line 122 is another potential source of bugs
The whole point of a state machine is to keep the states **isolated**

Each state should not need to be doing any sort of setup for the next. Let `createMenuPrompter` be called when the relevant prompt is used, not at the state transition. 

Looks like we don't need `defaultPrompter` either. The original 'RegionSubmenu' only exposed the active UI component as a mechanism for testing the control flow. But we're already exposing the factory functions for creating each prompt as public methods.